pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -701,6 +704,21 @@ type TarStrategy struct {
 // save/load the directory appropriately.
 type NoneStrategy struct{}
 
+// ArtifactLocationType is the type of artifact location
+type ArtifactLocationType string
+
+// ArtifactLocationType
+const (
```

Comment:
huh - I think this type already existed?

Reply:
Couldn't find it anywhere. Might have missed it though?